code/modules/mob/living/carbon/carbon.dm

```diff
@@ -859,43 +859,55 @@
+		var/edit_action = input(usr, "What would you like to do?","Modify Body Part") as null|anything in list("replace","remove")
 		if(!edit_action)
 			return
 		var/list/limb_list = list()
-		if(edit_action == "remove" || edit_action == "augment")
+		if(edit_action == "remove")
 			for(var/obj/item/bodypart/B in bodyparts)
 				limb_list += B.body_zone
-			if(edit_action == "remove")
 				limb_list -= BODY_ZONE_CHEST
 		else
-			limb_list = list(BODY_ZONE_HEAD, BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
-			for(var/obj/item/bodypart/B in bodyparts)
-				limb_list -= B.body_zone
-		var/result = input(usr, "Please choose which body part to [edit_action]","[capitalize(edit_action)] Body Part") as null|anything in limb_list
+			limb_list = list(BODY_ZONE_HEAD, BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG, BODY_ZONE_CHEST)
+		var/result = input(usr, "Please choose which bodypart to [edit_action]","[capitalize(edit_action)] Bodypart") as null|anything in sortList(limb_list)
 		if(result)
 			var/obj/item/bodypart/BP = get_bodypart(result)
-			switch(edit_action)
-				if("remove")
-					if(BP)
-						BP.drop_limb()
-					else
-						to_chat(usr, "[src] doesn't have such bodypart.")
-				if("add")
-					if(BP)
-						to_chat(usr, "[src] already has such bodypart.")
-					else
-						if(!regenerate_limb(result))
-							to_chat(usr, "[src] cannot have such bodypart.")
-				if("augment")
-					if(ishuman(src))
-						if(BP)
-							BP.change_bodypart_status(BODYTYPE_ROBOTIC, TRUE, TRUE)
-						else
-							to_chat(usr, "[src] doesn't have such bodypart.")
+			if(edit_action == "remove")
+				if(BP)
+					BP.drop_limb(TRUE)
+					admin_ticket_log("[key_name_admin(usr)] has removed [src]'s [parse_zone(BP.body_zone)]")
+				else
+					to_chat(usr, "<span class='boldwarning'>[src] doesn't have such bodypart.</span>")
+					admin_ticket_log("[key_name_admin(usr)] has attempted to modify the bodyparts of [src]")
+			else
+				var/list/limbtypes = list()
+				switch(result)
+					if(BODY_ZONE_CHEST)
+						limbtypes = typesof(/obj/item/bodypart/chest)
+					if(BODY_ZONE_R_ARM)
+						limbtypes = typesof(/obj/item/bodypart/r_arm)
+					if(BODY_ZONE_L_ARM)
+						limbtypes = typesof(/obj/item/bodypart/l_arm)
+					if(BODY_ZONE_HEAD)
+						limbtypes = typesof(/obj/item/bodypart/head)
+					if(BODY_ZONE_L_LEG)
+						limbtypes = typesof(/obj/item/bodypart/l_leg)
+					if(BODY_ZONE_R_LEG)
+						limbtypes = typesof(/obj/item/bodypart/r_leg)
+
+				if((edit_action == "add") && BP)
+					to_chat(usr, "<span class='boldwarning'>[src] already has such bodypart.</span>")
+				else
+					var/limb2add = input(usr, "Select a bodypart type to add", "Add/Replace Bodypart") as null|anything in sortList(limbtypes)
+					var/obj/item/bodypart/new_bp = new limb2add()
+
+					if(new_bp.replace_limb(src, TRUE, TRUE))
+						admin_ticket_log("key_name_admin(usr)] has replaced [src]'s [BP.type] with [new_bp.type]")
+						qdel(BP)
```

Comment:
Could have easily kept the Switch Statement.

Style wise, should have kept the actions together.